omod-2.2/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_2/DrugReferenceMapResource2_2.java

```diff
@@ -0,0 +1,142 @@
+import io.swagger.models.properties.StringProperty;
+
+@SubResource(path = "mapping", parent = DrugResource2_0.class, supportedClass = DrugReferenceMap.class, supportedOpenmrsVersions =
+{ "1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*", "2.2.*" })
+public class DrugReferenceMapResource2_2 extends DelegatingSubResource<DrugReferenceMap, Drug, DrugResource2_0> {
+	
+	@Override
+	public DrugReferenceMap newDelegate() {
+		return new DrugReferenceMap();
+	}
+	
+	@Override
+	public DrugReferenceMap save(DrugReferenceMap delegate) {
+		delegate.getDrug().addDrugReferenceMap(delegate);
+		Context.getConceptService().saveDrug(delegate.getDrug());
+		return delegate;
+	}
+	
+	@Override
+	public Drug getParent(DrugReferenceMap instance) {
+		return instance.getDrug();
+	}
+	
+	@Override
+	public void setParent(DrugReferenceMap instance, Drug parent) {
+		instance.setDrug(parent);
+	}
+	
+	@Override
+	public PageableResult doGetAll(Drug parent, RequestContext context) throws ResponseException {
+		List<DrugReferenceMap> maps = new ArrayList<DrugReferenceMap>(parent.getDrugReferenceMaps());
+		return new NeedsPaging<DrugReferenceMap>(maps, context);
+	}
+	
+	@Override
+	public DrugReferenceMap getByUniqueId(String uniqueId) {
+		
+		return Context.getService(RestHelperService.class).getObjectByUuid(DrugReferenceMap.class, uniqueId);
+	}
+	
+	@Override
+	protected void delete(DrugReferenceMap delegate, String reason, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+	
+	@Override
+	public void purge(DrugReferenceMap delegate, RequestContext context) throws ResponseException {
+		throw new ResourceDoesNotSupportOperationException();
+		
+	}
+	
+	@PropertyGetter("display")
+	public String getDisplayString(DrugReferenceMap map) {
+		if (map.getConceptMapType() == null) {
+			return "";
+		}
+		return map.getConceptMapType().getDescription();
+	}
+	
+	@PropertyGetter("conceptMapTypeName")
+	public String getconceptMapTypeName(DrugReferenceMap map) {
+		String name = map.getConceptMapType().getName();
+		return name;
+	}
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		if (rep instanceof DefaultRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("display");
+			description.addProperty("uuid");
+			description.addProperty("auditInfo");
+			description.addSelfLink();
+			description.addLink("full", ".?v=" + RestConstants.REPRESENTATION_FULL);
+			return description;
+		} else if (rep instanceof FullRepresentation) {
+			DelegatingResourceDescription description = new DelegatingResourceDescription();
+			description.addProperty("display");
+			description.addProperty("uuid");
+			description.addProperty("auditInfo");
+			description.addProperty("conceptMapTypeName");
+			description.addSelfLink();
+			return description;
+		}
+		return null;
+	}
```

Comment:
Code in line 105 to 108 and 113 to 116 is repeated. I suggest the repeated code to be placed before if structure.

Reply:
i get your point @haripriya999 , but this is like the standard way we do it in openmrs